libs/networking/http_helpers/http_helpers.ml

```diff
@@ -48,21 +58,35 @@ struct
            | None -> failwith "HTTP client not initialized")
     in
     let headers = Header.of_list headers in
-    let%lwt response, body = Client.get ~headers url in
-    let%lwt body = Cohttp_lwt.Body.to_string body in
+    let%lwt response, orig_body = Client.get ~headers url in
+    let%lwt body = Cohttp_lwt.Body.to_string orig_body in
     let code = response |> Response.status |> Code.code_of_status in
     match code with
-    | _ when Code.is_success code -> Lwt.return (Ok body)
+    | _ when Code.is_success code ->
+        Lwt.return (Ok (body, { code; response; body = orig_body }))
+    (* Automatically resolve redirects, in this case a 307 Temporary Redirect.
+       This is important for installing the Semgrep Pro Engine binary, which
+       receives a temporary redirect at the proper endpoint.
+    *)
+    | 307 -> (
```

Comment:
please add a comment in the .mli about get_async and the fact it handles 307 errors and automatically redirect